anomalib/deploy/inferencers/base.py

```diff
@@ -0,0 +1,172 @@
+            else:
+                meta_data = {}
+        if isinstance(image, (str, Path)):
+            image = read_image(image)
+        meta_data["image_shape"] = image.shape[:2]
+
+        processed_image = self.pre_process(image)
+        predictions = self.forward(processed_image)
+        anomaly_map, _ = self.post_process(predictions, meta_data=meta_data)
+
+        if superimpose is True:
+            anomaly_map = superimpose_anomaly_map(anomaly_map, image)
+
+        return anomaly_map
+
+    def __call__(self, image: np.ndarray) -> np.ndarray:
+        """Call predict on the Image.
+
+        Args:
+            image (np.ndarray): Input Image
+
+        Returns:
+            np.ndarray: Output predictions to be visualized
+        """
+        return self.predict(image)
+
+    def _normalize(
+        self,
+        anomaly_maps: Union[Tensor, np.ndarray],
+        pred_scores: Union[Tensor, np.float32],
+        meta_data: Union[Dict, DictConfig],
+    ) -> Tuple[Union[np.ndarray, Tensor], float]:
+        """Applies normalization and resizes the image.
+
+        Args:
+            anomaly_maps (Union[Tensor, np.ndarray]): Predicted raw anomaly map.
+            pred_scores (Union[Tensor, np.float32]): Predicted anomaly score
+            meta_data (Dict): Meta data. Post-processing step sometimes requires
+                additional meta data such as image shape. This variable comprises such info.
+
+        Returns:
+            Tuple[Union[np.ndarray, Tensor], float]: Post processed predictions that are ready to be visualized and
+                predicted scores.
+
+
+        """
+
+        # min max normalization
+        if "min" in meta_data and "max" in meta_data:
+            anomaly_maps = normalize_min_max(
+                anomaly_maps, meta_data["pixel_threshold"], meta_data["min"], meta_data["max"]
+            )
+            pred_scores = normalize_min_max(
+                pred_scores, meta_data["image_threshold"], meta_data["min"], meta_data["max"]
+            )
+
+        # standardize pixel scores
+        if "pixel_mean" in meta_data.keys() and "pixel_std" in meta_data.keys():
+            anomaly_maps = standardize(
+                anomaly_maps, meta_data["pixel_mean"], meta_data["pixel_std"], center_at=meta_data["image_mean"]
+            )
+            anomaly_maps = normalize_cdf(anomaly_maps, meta_data["pixel_threshold"])
+
+        # standardize image scores
+        if "image_mean" in meta_data.keys() and "image_std" in meta_data.keys():
+            pred_scores = standardize(pred_scores, meta_data["image_mean"], meta_data["image_std"])
+            pred_scores = normalize_cdf(pred_scores, meta_data["image_threshold"])
+
+        if isinstance(anomaly_maps, Tensor):
+            anomaly_maps = anomaly_maps.cpu().numpy()
+
+        if "image_shape" in meta_data and anomaly_maps.shape != meta_data["image_shape"]:
+            anomaly_maps = cv2.resize(anomaly_maps, meta_data["image_shape"])
```

Comment:
I feel that the Numpy conversion and resizing should be taken outside of the normalize method, because both these operations are not related to normalization.